exotica_core/include/exotica_core/kinematic_tree.h

```diff
@@ -210,6 +210,9 @@ class KinematicTree : public Uncopyable
 
     std::map<std::string, shapes::ShapeType> GetCollisionObjectTypes();
 
+     /// @brief SetSeed set the seed of the random generator for deterministic joint state sampling
+     /// @param seed unsigned integer
+    void SetSeed(const uint_fast32_t seed) { generator_.seed(seed); }
```

Comment:
`const uint_fast32_t&` (pass by reference, minor here but let's be consistent)

Reply:
What's the advantage of using a reference here? `uint32` and its reference have the same size, i.e. there is no performance benefit.

Reply:
You're right and I missed something here. We shouldn't do it for fundamental types (cf. [here](http://www.learncpp.com/cpp-tutorial/73-passing-arguments-by-reference/)) and apart from double and long long the types are smaller than their reference (cf. [here](https://softwareengineering.stackexchange.com/a/356200)). Also apparently it may break compiler optimisation by aliasing (duh). Sorry I didn't realise that earlier, let's go back to how you initially had it and apologies for the extra trouble :'(.

Reply:
No problem :-)